subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/initialization/InstantExecutionStartParameter.kt

```diff
@@ -81,7 +82,8 @@ class InstantExecutionStartParameter(
         // and can be used as separators: /, \, :, <, >, ", ?, *, |
         // except we also accept qualified task names with :, so colon is out.
         val cacheKey = StringBuilder()
-        requestedTaskNames.joinTo(cacheKey, separator = "/")
+        cacheKey.append(GradleVersion.current().version)
```

Comment:
Now that we're using a cross version cache, we should probably use `Hasher` to assemble the hash for the key. Probably should also add an initial cross version test to verify that different Gradle versions do not interfere with each other.

It's also kind of odd that the start parameter knows how to calculate the cache entry key.